sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/EncryptingDataBase.cs

```diff
@@ -0,0 +1,66 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class EncryptingDataBase<TSyntaxKind> : SonarDiagnosticAnalyzer
+        where TSyntaxKind : struct
+    {
+        protected const string DiagnosticId = "S4787";
+        protected const string MessageFormat = "Make sure that encrypting data is safe here.";
+
+        private static readonly ImmutableArray<KnownType> forbiddenBaseTypes =
```

Comment:
I used _ImmutableArray_ here since that is what is expected by the existing helper method _DerivesFromAny_.
I considered changing the helper method signature to use _IEnumerable_, but that would have been inconsistent with the other helper methods in the class and I didn't want to change them all.

Reply:
Also ImmutableArray<T> is a struct and changing the parameter to IEnumerable<T> would cause boxing...